### PR TITLE
Make named jar use jetbrain annotations

### LINKED ISSUE #16: Annotation replacement
https://github.com/FabricMC/fabric-loom/blob/57c9a8f3209121f22e84b8b01b64f1093256f1d3/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftMappedProvider.java#L46-L50

Should be easy to do. Also avoids tricky bad javadoc problems from jsr305 annotations

### PR DESCRIPTION
**Pull request by liach**
and subsequent javadoc tweaks

Closes FabricMC/MappingPoet#16

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -817,16 +828,21 @@ publishing {
 	}
 }
 
-
 void mapJar(File output, File input, File mappings, File libraries, String from, String to) {
+	mapJar(output, input, mappings, libraries, from, to, { })
+}
+
+void mapJar(File output, File input, File mappings, File libraries, String from, String to, Action<TinyRemapper.Builder> action) {
```
Comment (modmuss50) [inline build.gradle]: I think you can just do:

```groovy
void mapJar(File output, File input, File mappings, File libraries, String from, String to, Action<TinyRemapper.Builder> action = { }) {
```

**Review by modmuss50 — APPROVED** [review]
Looks fine, just that minor nitpick

File: build.gradle (hunk 1 above)
Comment (liach) [reply build.gradle]: hmm, groovy has default args? let me try

Comment (liach) [reply build.gradle]: It works perfectly! wrapped the line for the action arg because otherwise the line feels too long

Comment (modmuss50) [reply build.gradle]: Yeah, groovy has a lot of little tricks like this.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**